RetailCoder.VBE/Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs

```diff
@@ -0,0 +1,229 @@
+            {
+                var message = string.Format(RubberduckUI.MoveCloserToUsage_TargetHasNoReferences, target.IdentifierName);
+
+                _messageBox.Show(message, RubberduckUI.MoveCloserToUsage_Caption, System.Windows.Forms.MessageBoxButtons.OK,
+                    System.Windows.Forms.MessageBoxIcon.Exclamation);
+
+                return;
+            }
+
+            if (TargetHasReferencesOnlyInOneMethod(target))
+            {
+                var message = string.Format(RubberduckUI.MoveCloserToUsage_TargetIsUsedInMultipleMethods, target.IdentifierName);
+                _messageBox.Show(message, RubberduckUI.MoveCloserToUsage_Caption, System.Windows.Forms.MessageBoxButtons.OK,
+                    System.Windows.Forms.MessageBoxIcon.Exclamation);
+
+                return;
+            }
+
+            MoveDeclaration(target);
+        }
+
+        private bool TargetHasReferencesOnlyInOneMethod(Declaration target)
+        {
+            var firstReference = target.References.FirstOrDefault();
+
+            return firstReference != null && target.References.Any(r => r.ParentScope != firstReference.ParentScope);
+        }
+
+        private void MoveDeclaration(Declaration target)
+        {
+            InsertDeclaration(target);
+            RemoveVariable(target);
+        }
+
+        private void InsertDeclaration(Declaration target)
+        {
+            var firstReference = target.References.OrderBy(r => r.Selection.StartLine).First();
+
+            var oldLines = _editor.GetLines(firstReference.Selection);
+            var newLines = oldLines.Insert(firstReference.Selection.StartColumn - 1, GetDeclarationString(target));
+
+            _editor.DeleteLines(firstReference.Selection);
+            _editor.InsertLines(firstReference.Selection.StartLine, newLines);
+        }
+
+        private string GetDeclarationString(Declaration target)
+        {
+            return Environment.NewLine + "    Dim " + target.IdentifierName + " As " + target.AsTypeName + Environment.NewLine;
+        }
+
+        private void RemoveVariable(Declaration target)
+        {
+            Selection selection;
+            var declarationText = target.Context.GetText();
+            var multipleDeclarations = HasMultipleDeclarationsInStatement(target);
+
+            var variableStmtContext = GetVariableStmtContext(target);
+
+            if (!multipleDeclarations)
+            {
+                declarationText = variableStmtContext.GetText();
+                selection = GetVariableStmtContextSelection(target);
+            }
+            else
+            {
+                selection = new Selection(target.Context.Start.Line, target.Context.Start.Column,
+                    target.Context.Stop.Line, target.Context.Stop.Column);
+            }
+
+            var oldLines = _editor.GetLines(selection);
+
+            var newLines = oldLines.Replace(" _" + Environment.NewLine, string.Empty)
```

Comment:
That replace here is fishy. It assumes you instantly newline after a line-continuation. I don't know how Rubberduck handles the editor's lines, but I wouldn't assume that.